src/GI.h

```diff
@@ -273,8 +340,34 @@ template <class Ptcl> class GI : public Problem<Ptcl>{
                     ptcl.push_back(imp[i]);
                 }
                 break;
+	      }
         }
 
+	        std::cout << "Target  :" << tarNptcl << std::endl;
+		std::cout << "    radius           : " << tarRadi << std::endl;
+		std::cout << "    total-to-core    : " << (double)(tarNcore) / (double)(tarNptcl) << std::endl;
+		std::cout << "    # of core ptcls  : " << tarNcore << std::endl;
+		std::cout << "    # of mantle ptcls: " << tarNmntl << std::endl;
+		std::cout << "    core density     : " << tarCoreMass / (4.0 * math::pi / 3.0 * tarCoreRadi * tarCoreRadi * tarCoreRadi * Corr * Corr * Corr) << std::endl;
+		std::cout << "    mantle density   : " << (tarMass - tarCoreMass) / (4.0 * math::pi / 3.0 * (tarRadi * tarRadi * tarRadi - tarCoreRadi * tarCoreRadi * tarCoreRadi)) << std::endl;
+		std::cout << "    mean density     : " << tarMass / (4.0 * math::pi / 3.0 * tarRadi * tarRadi * tarRadi) << std::endl;
+
+		if (mode==1){
+		  std::cout << "Impactor:" << impNptcl << std::endl;
+		  std::cout << "    radius           : " << impRadi << std::endl;
+		  std::cout << "    total-to-core    : " << (double)(impNcore) / (double)(impNptcl) << std::endl;
+		  std::cout << "    # of core ptcls  : " << impNcore << std::endl;
+		  std::cout << "    # of mantle ptcls: " << impNmntl << std::endl;
+		  std::cout << "    core density     : " << impCoreMass / (4.0 * math::pi / 3.0 * impCoreRadi * impCoreRadi * impCoreRadi * Corr * Corr * Corr) << std::endl;
+		  std::cout << "    mantle density   : " << (impMass - impCoreMass) / (4.0 * math::pi / 3.0 * (impRadi * impRadi * impRadi - impCoreRadi * impCoreRadi * impCoreRadi)) << std::endl;
+		  std::cout << "    mean density     : " << impMass / (4.0 * math::pi / 3.0 * impRadi * impRadi * impRadi) << std::endl;
+		  std::cout << "    velocity         : " << impVel << std::endl;
+		  std::cout << "Tar-to-Imp mass ratio: " << (double)(impNmntl) / (double)(tarNmntl) << std::endl;
+		}
+		
+		std::cout << "Total:" << Nptcl << std::endl;
```

Comment:
Please add the following:
```suggestion
		assert(Nptcl == tarNptcl + impNptcl);
		std::cout << "Total:" << Nptcl << std::endl;
```
I am a bit worried that Nptcl and tarNptcl + impNptcl can become inconsistent, because we determine them in different ways. You may need to include `#include <assert.h>` at the top of the file for this.